scipy/stats/_distn_infrastructure.py

```diff
@@ -3245,6 +3198,103 @@ def _iter_chunked(x0, x1, chunksize=4, inc=1):
         yield supp
 
 
+class rv_sample(rv_discrete):
+    """A 'sample' discrete distribution defined by the support and values.
+
+       The ctor ignores most of the arguments, only needs the `values` argument.
+    """
+    def __init__(self, a=0, b=inf, name=None, badvalue=None,
+                 moment_tol=1e-8, values=None, inc=1, longname=None,
+                 shapes=None, extradoc=None, seed=None):
+
+        super(rv_discrete, self).__init__(seed)
+
+        if values is None:
+            raise ValueError("rv_sample.__init__(..., values=None,...)")
+
+        # cf generic freeze
+        self._ctor_param = dict(
+            a=a, b=b, name=name, badvalue=badvalue,
+            moment_tol=moment_tol, values=values, inc=inc,
+            longname=longname, shapes=shapes, extradoc=extradoc, seed=seed)
+
+        if badvalue is None:
+            badvalue = nan
+        self.badvalue = badvalue
+        self.moment_tol = moment_tol
+        self.inc = inc
+        self.shapes = shapes
+        self.vecentropy = self._entropy
+
+        self.xk, self.pk = values
```

Comment:
this can be ``xk, pk = values``, make this and the next few lines shorter and clearer